src/rsa.rs

```diff
@@ -244,6 +255,49 @@ impl RSAKeyPair {
     pub fn public_modulus_len(&self) -> usize {
         unsafe { RSA_size(self.rsa.as_ref()) }
     }
+
+    /// Sign the given message, hashed with the given digest algorithm.
+    pub fn sign(&self, digest_alg: &'static digest::Algorithm, msg: &[u8], signature: &mut [u8]) -> Result<(), ()> {
+        if signature.len() != self.public_modulus_len() {
+            return Err(());
+        }
+
+        let found = PREFIXES.into_iter().find(|pair| pair.0.nid == digest_alg.nid);
+        if found.is_none() {
+            return Err(());
+        }
+
+        let msg_len = self.public_modulus_len();
+        let hashed = digest::digest(digest_alg, msg);
+        let prefix = found.unwrap().1;
+        let digest_len = prefix.len() + hashed.as_ref().len();
+        if msg_len < digest_len + 11 {
+            return Err(());
+        }
+
+        let padding_len = msg_len - digest_len - 3;
+        let mut pad_bytes: Vec<u8> = Vec::with_capacity(msg_len);
+        pad_bytes.push(0);
+        pad_bytes.push(1);
+        for _ in 0..padding_len {
+            pad_bytes.push(255);
+        }
+        pad_bytes.push(0);
+        pad_bytes.extend(prefix);
+        pad_bytes.extend(hashed.as_ref());
+
+        let rng = rand::SystemRandom::new();
+        let mut rand = RAND::new(&rng);
+        try!(bssl::map_result(unsafe {
+            let blinding = BN_BLINDING_new();
```

Comment:
This is OK for the first attempt. But, this is very inefficient. Besides the memory allocation that `BN_BLINDING_new()` does, the first time you *use* a `BN_BLINDING` is much more expensive than the subsequent times. So, later, we should store `blinding` in the `RSAPrivateKeyPair` struct and reuse it (using a mutex).